opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -270,10 +312,33 @@ def create_metric(
             description,
             unit,
             value_type,
+            self,
             label_keys=label_keys,
             enabled=enabled,
             monotonic=monotonic,
         )
 
+    def get_label_set(self, labels: Dict[str, str]):
+        """See `opentelemetry.metrics.Meter.create_metric`.
+
+        This implementation encodes the labels to use as a map key.
+
+        Args:
+            labels: The dictionary of label keys to label values.
+        """
+        if len(labels) == 0:
+            return EMPTY_LABEL_SET
+        sorted_labels = OrderedDict(sorted(labels.items()))
+        # Uses statsd encoding for labels
+        encoded = "|#" + ",".join(
+            "%s:%s" % (key, value) for (key, value) in sorted_labels.items()
```

Comment:
I've changed the implementation to follow the [Go default implementation](https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/metric/labelencoder.go#L48) of the label encoder (although it will still have the same unambiguous problem with key/values with '=' in it). I am not sure if it is a safe assumption to assume that the "=" will not occur. Thoughts?

Reply:
Maybe rise this issue in the Go repository? Or ask in the spec repository? My assumption would be that while we might want to restrict the key strings, we should not restrict values to not contain e.g. `|#`.

Reply:
I've filed an issue [here](https://github.com/open-telemetry/opentelemetry-go/issues/332)